Sources/OpenPass/Data/OpenPassTokens.swift

```diff
@@ -28,38 +28,134 @@ import Foundation
+    }
+
+    /// When the ID Token expires. If nil, the expiry is unknown
+    public var idTokenExpiry: Date? {
+        guard let idTokenExpiresIn else {
+            return nil
+        }
+        return issuedAt.map { $0.addingTimeInterval(TimeInterval(idTokenExpiresIn)) }
+    }
     
+    /// When the Refresh Token expires. If nil, the expiry is unknown
+    public var refreshTokenExpiry: Date? {
+        guard let refreshTokenExpiresIn else {
+            return nil
+        }
+        return issuedAt.map { $0.addingTimeInterval(TimeInterval(refreshTokenExpiresIn)) }
+    }
+}
+
+extension OpenPassTokens {
     /// Primary Constructor
     /// - Parameters:
     ///   - idTokenJWT: ID Token as JWT
     ///   - accessToken: Access Token
     ///   - tokenType: Type of Access Token
-    ///   - expiresIn: Exipiration time of token, represented in UTC
-    public init(idTokenJWT: String, accessToken: String, tokenType: String, expiresIn: Int64) {
-        self.idTokenJWT = idTokenJWT
-        self.accessToken = accessToken
-        self.tokenType = tokenType
-        self.expiresIn = expiresIn
-
-        self.idToken = IDToken(idTokenJWT: idTokenJWT)
+    ///   - expiresIn: Seconds until the Access Token expires
+    ///   - refreshToken: Refresh Token
+    ///   - refreshTokenExpiresIn: Seconds until the Refresh Token expires
+    ///   - issuedAt: Date when the tokens were issued
+    public init(
+        idTokenJWT: String,
+        idTokenExpiresIn: Int64?,
+        accessToken: String,
+        tokenType: String,
+        expiresIn: Int64,
+        refreshToken: String?,
+        refreshTokenExpiresIn: Int64?,
+        issuedAt: Date
+    ) {
+        self.init(
+            idToken: IDToken(idTokenJWT: idTokenJWT),
+            idTokenJWT: idTokenJWT,
+            idTokenExpiresIn: idTokenExpiresIn,
+            accessToken: accessToken,
+            tokenType: tokenType,
+            expiresIn: expiresIn,
+            refreshToken: refreshToken,
+            refreshTokenExpiresIn: refreshTokenExpiresIn,
+            issuedAt: issuedAt
+        )
+    }
+}
+
+extension OpenPassTokens {
+
+    /// A convenience initializer for processing a Token response from ``OpenPassClient``
+    /// The current date should be passed as now to be used as an `issuedAt` value if the response does not contain one.
+    init(_ response: OpenPassTokensResponse, now: Date = .init()) throws {
+        switch response {
+        case .success(let tokens):
+            guard let idToken = tokens.idToken,
+                  let expiresIn = tokens.expiresIn else {
+                throw OpenPassError.tokenData(
+                    name: "OpenPassToken Generator",
+                    description: "Unable to generate OpenPassTokens from server",
+                    uri: nil
+                )
+            }
+
+            let issuedAt: Date
+            if let responseIssuedAt = tokens.issuedAt {
```

Comment:
I don't believe the API can ever return `issueAt`, can it? Even if it could, you can't guarantee that the clocks would match between the two, so your local `now` is more reliable.